src/main/java/hudson/plugins/gradle/GradleInstallation.java

```diff
@@ -111,17 +108,16 @@ public String getDisplayName() {
             return Collections.singletonList(new GradleInstaller(null));
         }
 
-        // for compatibility reasons, the persistence is done by GradleBuilder.DescriptorImpl
+        // for compatibility reasons, the persistence is done by Gradle.DescriptorImpl
 
         @Override
         public GradleInstallation[] getInstallations() {
-            return gradleDescriptor.getInstallations();
+            return Jenkins.getActiveInstance().getDescriptorByType(Gradle.DescriptorImpl.class).getInstallations();
```

Comment:
Can we keep the injected field and use the static lookup only if the field is `null`? This logic should probably also extracted into a `getGradleDescriptor` method.